asf_levies_model/consumers.py

```diff
@@ -0,0 +1,460 @@
+        scheme_eligible: Eligibilty for social support scheme.
+
+    Properties:
+        unit_cost_ratio: Electricity to gas unit cost ratio, considering any unit discounts from social support.
+        fuel_poverty_gap: Difference between total fuel spend and threshold of 10% of net annual income, in £.
+        electricity_subtotal_bill: Electricity subtotal bill, including VAT, before any social support adjustment, in £.
+        gas_subtotal_bill: Gas subtotal bill, including VAT, before any social support adjustment, in £.
+        electricity_bill: Electricity final bill, including VAT, after any social support adjustment, in £.
+        gas_bill: Gas final bill, including VAT, after any social support adjustment, in £.
+        dual_fuel_bill: Combined electricity and gas final bill, including VAT, after any social support adjustment, in £.
+
+    Methods:
+        apply_social_support_adjustment():
+            Overwrites the electricity or gas bill property with an updated value reflected a social support adjustment (discount or additional cost).
+        get_tidy_summary():
+            Returns a dataframe containing all public instance attributes and properties, except name which has a dedicated column.
+        consumer_from_dataframe():
+            Creates a Consumer instance from a dataframe input.
+        boiler_to_hp_savings():
+            Returns a cost savings estimate if consumer is assumed to switch from a gas boiler for heat to an electric heat pump.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        archetype: str,
+        net_annual_income: float,
+        net_income_decile: int,
+        main_heating_fuel: str,
+        gas_consumption: float,
+        electricity_consumption: float,
+        gas_tariff: object,
+        electricity_tariff: object,
+        unmetered_fuel_spend: float = 0,
+        scheme_eligible: bool = False,
+        size: int = 0,
+    ) -> None:
+        """Initalizes a Consumer instance based on provided parameters.
+
+        Parameters
+        ----------
+        name : str
+            Unique name of the energy consumer.
+        archetype : str
+            Name of broader archetype the consumer belongs to.
+        net_annual_income : float
+            The average net annual income of represented consumer, in £.
+        net_income_decile : int
+            The income decile that the consumer's net annual income falls in.
+        main_heating_fuel : str
+            The name of the consumer's main heating fuel.
+        gas_consumption : float
+            Annual gas consumption, in MWh.
+        electricity_consumption : float
+            Annual electricity consumption, in MWh.
+        gas_tariff : object
+            Tariff object to be used to calculate the consumer's gas bill.
+        electricity_tariff : object
+            Tariff object to be used to calculate the consumer's electricity bill.
+        unmetered_fuel_spend : float, optional
```

Comment:
gas and electricity tariffs are Tariff objects, so we can import these and reference them specifically.